powerdns/admin.py

```diff
@@ -52,12 +56,18 @@ def queryset(self, request, queryset):
 
 
 class RecordAdminForm(ModelForm):
+
     def clean_type(self):
         type = self.cleaned_data['type']
         if not type:
             raise ValidationError(_("Record type is required"))
         return type
 
+    def clean_domain(self):
+        validator = PermissionValidator('powerdns.change_domain')
```

Comment:
ad 1. In most cases the user is set in 'set_context' method of UserBasedValidator
ad 2. You can go with superusers only